### PR TITLE
update to SOS 1.3.2 and try OOP build

### PR DESCRIPTION
**Pull request by jeffhammond**
do not merge until Travis CI confirms this works...

### CODE REVIEW

File: travis/install-sandia-openshmem.sh [hunk 1]
```diff
@@ -18,17 +18,14 @@ if [ ! -d "$SHMEM_ROOT" ]; then
...
+    # 1.3.2 release
+    wget https://github.com/Sandia-OpenSHMEM/SOS/archive/v1.3.2.tar.gz
+    tar -xzf v1.3.2.tar.gz
+    cd SOS-1.3.2
     ./autogen.sh
-    # must build in-place (https://github.com/regrant/sandia-shmem/issues/49)
-    ./configure --with-ofi=$TRAVIS_ROOT/libfabric --with-ofi-libdir=$TRAVIS_ROOT/libfabric/lib \
+    mkdir build
+    cd build
+    ../configure --with-ofi=$TRAVIS_ROOT/libfabric --with-ofi-libdir=$TRAVIS_ROOT/libfabric/lib \
```
Comment (jdinan) [inline travis/install-sandia-openshmem.sh]: Please change to ```--with-libfabric=$TRAVIS_ROOT/libfabric``` and delete ```--with-ofi-libdir=$TRAVIS_ROOT/libfabric/lib```

Comment (jdinan) [inline travis/install-sandia-openshmem.sh]: Please also remove ```--with-pmi=$TRAVIS_ROOT/hydra```.  This is not needed if you enable simple PMI.

File: travis/install-sandia-openshmem.sh [hunk 2]
```diff
@@ -25,12 +24,13 @@ if [ ! -d "$SHMEM_ROOT" ]; then
     ./autogen.sh
     mkdir build
     cd build
-    ../configure --with-ofi=$TRAVIS_ROOT/libfabric --with-ofi-libdir=$TRAVIS_ROOT/libfabric/lib \
-                --disable-fortran \
-                --enable-error-checking \
-                --enable-remote-virtual-addressing \
-                --enable-pmi-simple --with-pmi=$TRAVIS_ROOT/hydra \
-                --prefix=$SHMEM_ROOT
+    ../configure --with-ofi=$TRAVIS_ROOT/libfabric \
```
Comment (jdinan) [inline travis/install-sandia-openshmem.sh]: The ```--with-ofi``` option was deprecated and replaced with ```--with-libfabric```, so you can safely remove ```--with-ofi```.

File: travis/install-sandia-openshmem.sh (hunk 1 above)
Comment (jeffhammond) [inline travis/install-sandia-openshmem.sh]: I must be super-stupid, because it is failing with `oshrun: Error, could not find a launcher.  Aborting.`

**Comment by jdinan** [comment]
Were you able to sort out the launcher issue?  Right now, none of the configure settings actually touch the ```oshrun``` wrapper.  Would it be helpful if we add a configure setting to set the location of the ```oshrun``` launcher?

**Comment by jeffhammond** [comment]
No. Have not solved. Will just revert to old PMI config opts and see if that works.

**Comment by jdinan** [comment]
If the problem is with oshrun finding the launcher, that's not going to help.  The configure args have no effect on the oshrun script.

It looks like the problem is that you removed Hydra, but haven't replaced it with a different launcher.

**Comment by jeffhammond** [comment]
Ugh. Why did you tell me to change anything then? It wasn't broken. "Fixing" it broke it.

Your comments suggested simple PMI was sufficient. I don't see any value in any changes related to PMI if I still need Hydra.

Will just revert everything and start over.

**Comment by jdinan** [comment]
Not sure how you got that from my comment.  The changes to the SOS configure args and removal of LD_LIBRARY_PATH updates look good.

**Comment by jeffhammond** [comment]
@jdinan The comment that confused me was:
> Please also remove --with-pmi=$TRAVIS_ROOT/hydra. This is not needed if you enable simple PMI.
So I do not need the configure option but I still need Hydra?  I read your comment as "you do not need Hydra if you enable simple PMI".

**Comment by jdinan** [comment]
You need Hydra.  If you give --enable-simple-pmi, you should not also provide --with-pmi.  Those arguments are in conflict; each is telling SOS to use a different PMI client library setting.

**Comment by jeffhammond** [comment]
Thanks.  That clears things up.

**Pull request closed without merging**